x-pack/test/api_integration/services/infraops_graphql_client.js

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import ApolloClient from 'apollo-boost'; // eslint-disable-line 
+import { format as formatUrl } from 'url';
+import fetch from 'node-fetch';
+
+export function InfraOpsGraphQLProvider({ getService }) {
+  const config = getService('config');
+  const kbnURL = formatUrl(config.get('servers.kibana'));
+
+  return new ApolloClient({
+    uri: `${kbnURL}/api/infra/graphql`,
+    fetch,
+    request: (operation) => {
+      operation.setContext({
+        headers: {
+          'kbn-xsrf': 'xxx'
+        }
+      });
+    }
+  });
```

Comment:
...and configure the `ApolloClient` like this:

```typescript
  return new ApolloClient({
    cache: new InMemoryCache(),
    link: new HttpLink({
      credentials: 'same-origin',
      fetch,
      headers: {
        'kbn-xsrf': 'xxx',
      },
      uri: `${kbnURL}/api/infra/graphql`,
    }),
  });
```

Reply:
Done